### PR TITLE
Re-organize Markdown files

### PR DESCRIPTION
**Pull request by michaelbaudino**
This commit hopefully makes our Markdown files more readable by:

* moving all conventions to the `docs` subdirectory (no more in README)
* move inline link targets down to a Bibliography section
* reword some sentences for clarity and explicitness

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,67 +17,56 @@ Code style used by AlpineLab in all our projects.
...
-      gem "alpinelab-codestyle", "~> x.y"
-    end
-    ```
-
-    or install it manually:
-
-    ```shell
-    gem install alpinelab-codestyle
-    ```
-
```
Comment (ghost) [inline README.md:36]: Merging those sections is a little out of the scope of this PR, but I think
it's acceptable.

File: docs/rails.md [hunk 2]
```diff
@@ -75,3 +81,6 @@ Rails.application.routes.draw do
   end
 end
 ```
+
+<!-- Bibliography -->
+[12-factor config]: https://12factor.net/config
```
Comment (tjouan) [inline docs/rails.md]: I think most changes in this file are out of scope of this PR, and that we
actually lost some information, and added a little confusion.
I would extract them in a new PR (or multiple ones) so that we can discuss
them.

**Review by ghost — CHANGES_REQUESTED** [review]
.

**Comment by michaelbaudino** [comment]
@tjouan you were right on many thing, the main one being that this PR included too many unrelated changes. I extracted it to multiple new PRs (#51, #52, #53, #54 and #55) and I can this close this very PR.
